lib/page/search_events.dart

```diff
@@ -96,20 +100,15 @@ class _BuildSearchEventsState extends State<BuildSearchEvents> {
               hasSecondIcon: true,
               secondIcon: const Icon(Icons.filter_list),
               iconOnPressedHandler: () async {
-                await showModalBottomSheet(
-                    context: context,
-                    shape: const RoundedRectangleBorder(
-                      borderRadius: BorderRadius.vertical(
-                        top: Radius.circular(CustomRadius.body),
-                      ),
-                    ),
-                    builder: (context) => FilterModal(
-                          filter: filter,
-                        )).then((value) => setState(() {
-                      if (value != null) {
-                        filter = value;
-                      }
-                    }));
+                await Navigator.of(context, rootNavigator: true)
+                    .push(MaterialPageRoute(
+                        builder: (context) =>
+                            SearchEventsFilter(filter: filter)))
+                    .then((value) => setState(() {
+                          if (value != null) {
+                            filter = value;
+                          }
+                        }));
                 // ignore: use_build_context_synchronously
```

Comment:
Dont ignore this linter warning. refer to [this](https://stackoverflow.com/questions/69466478/waiting-asynchronously-for-navigator-push-linter-warning-appears-use-build)